seissolxdmfwriter/seissolxdmfwriter/seissol_output_extractor.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python3
+import os
+import os.path
+import seissolxdmf
+import seissolxdmfwriter as sxw
+import numpy as np
+import argparse
+
+
+def generate_new_prefix(prefix, append2prefix):
+    prefix = os.path.basename(prefix)
+    lsplit = prefix.split("-")
+    if len(lsplit) > 1:
+        if lsplit[-1] in ["surface", "low", "fault"]:
+            prefix0 = "-".join(lsplit[0:-1])
+            prefix_new = prefix0 + append2prefix + "-" + lsplit[-1]
+        else:
+            prefix_new = prefix + append2prefix
+    else:
+        prefix_new = prefix + append2prefix
+    return prefix_new
+
+
+parser = argparse.ArgumentParser(
+    description="Extracts and processes data from SeisSol output files"
+)
+parser.add_argument("xdmfFilename", help="SeisSol XDMF output filename")
+parser.add_argument(
+    "--add2prefix",
+    help="string to append to the prefix in the new file",
+    type=str,
+    default="_resampled",
+)
+parser.add_argument(
+    "--variables",
+    nargs="+",
+    metavar="variables",
+    help="Names of variables to extract (e.g. SRs or all)",
+    default=["all"],
+)
+parser.add_argument(
+    "--precision",
+    type=str,
+    choices=["float", "double"],
+    default="float",
+    help="precision of the data in the output file",
+)
+parser.add_argument(
+    "--backend",
+    type=str,
+    choices=["hdf5", "raw"],
```

Comment:
raw -> posix. At least that's the termionology in the SeisSol parameter file.

Reply:
See comment of Lukas:
https://github.com/SeisSol/SeisSol/pull/297#discussion_r580114139